### PR TITLE
Add MarkAnsweredServlet and tests

### PR DESCRIPTION
**Pull request by n-wach**
Adds a new Servlet, and updates the Comment.Type enum (which involves changing some client code too).

Closes: #334 
Closes: #333

### CODE REVIEW

File: src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  @Override
+  public void init() throws ServletException {
+    userService = UserServiceFactory.getUserService();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Optional<String> error = validatePostRequest(request);
+    if (error.isPresent()) {
```
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java:62]: Would it be possible to create helper methods for the various checks in this method?

File: src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  private DatastoreService datastore;
+
+  @Override
+  public void init() throws ServletException {
+    userService = UserServiceFactory.getUserService();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (jaysaleh) [inline src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java:60]: doPost is doing too much you might want to decompose

File: src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java (hunk 1 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java:62]: Done!

File: src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java (hunk 2 above)
Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java:60]: Done!

File: src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java [hunk 3]
```diff
@@ -110,4 +91,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     return Optional.empty();
   }
+
+  /**
+   * Updates the question-type comment specified by `commentKey` to `newType`.  Returns
+   * {@Optional.empty()} if there were no errors, otherwise returns an error message.
```
Comment (amuamushu) [inline src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java:97]: Is this supposed to be {@code Optional.empty()}?

Comment (n-wach) [reply src/main/java/com/googleinterns/zoomtube/servlets/MarkAnsweredServlet.java:97]: ah yes, thanks for catching that

**Pull request merged**